src/test/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/prisonersearch/PrisonerSearchServiceTest.kt

```diff
@@ -54,76 +56,10 @@ class PrisonerSearchServiceTest {
     }
   }
 
-  @Nested
-  inner class Create_PrisonerSearchRequest {
-    @Test
-    fun `should create PrisonerSearchRequest given BarcodeRecipient with prisonNumber`() {
-      val barcodeRecipient = BarcodeRecipient(
-        barcode = aBarcode(),
-        prisonCode = "BXI",
-        prisonNumber = "A1234BC",
-        name = "John Smith",
-        dob = null
-      )
-      val expectedPrisonSearchRequest =
-        PrisonerSearchRequest(prisonNumber = "A1234BC", firstName = "John", lastName = "Smith", dob = null)
-
-      val prisonerSearchRequest = prisonerSearchRequest(barcodeRecipient)
-
-      assertThat(prisonerSearchRequest).isEqualTo(expectedPrisonSearchRequest)
-    }
-
-    @Test
-    fun `should create PrisonerSearchRequest given BarcodeRecipient with DOB`() {
-      val barcodeRecipient = BarcodeRecipient(
-        barcode = aBarcode(),
-        prisonCode = "BXI",
-        prisonNumber = null,
-        name = "John Smith",
-        dob = LocalDate.MIN
-      )
-      val expectedPrisonSearchRequest =
-        PrisonerSearchRequest(prisonNumber = null, firstName = "John", lastName = "Smith", dob = LocalDate.MIN)
-
-      val prisonerSearchRequest = prisonerSearchRequest(barcodeRecipient)
-
-      assertThat(prisonerSearchRequest).isEqualTo(expectedPrisonSearchRequest)
-    }
-
-    @Test
-    fun `should create PrisonerSearchRequest given BarcodeRecipient with name with many components`() {
-      val barcodeRecipient = BarcodeRecipient(
-        barcode = aBarcode(),
-        prisonCode = "BXI",
-        prisonNumber = "A1234BC",
-        name = "John Bobby Smith",
-        dob = null
-      )
-      val expectedPrisonSearchRequest =
-        PrisonerSearchRequest(prisonNumber = "A1234BC", firstName = "John", lastName = "Bobby Smith", dob = null)
-
-      val prisonerSearchRequest = prisonerSearchRequest(barcodeRecipient)
-
-      assertThat(prisonerSearchRequest).isEqualTo(expectedPrisonSearchRequest)
-    }
-
-    @Test
-    fun `should create PrisonerSearchRequest given BarcodeRecipient with name with just one component`() {
-      val barcodeRecipient =
-        BarcodeRecipient(barcode = aBarcode(), prisonCode = "BXI", prisonNumber = "A1234BC", name = "John", dob = null)
-      val expectedPrisonSearchRequest =
-        PrisonerSearchRequest(prisonNumber = "A1234BC", firstName = null, lastName = "John", dob = null)
-
-      val prisonerSearchRequest = prisonerSearchRequest(barcodeRecipient)
-
-      assertThat(prisonerSearchRequest).isEqualTo(expectedPrisonSearchRequest)
-    }
-  }
-
   private fun aBarcode() = Barcode("SOME_BARCODE")
 
-  private fun aMatchPrisonersResponse() = MatchPrisonersResponse(
-    matchedBy = ALL_SUPPLIED,
+  private fun aMatchPrisonersResponse() = PrisonerMatches(
+    matchedBy = aLLSUPPLIED,
```

Comment:
Typo or bizarre return value? 😆

Reply:
No, this is (unfortunately) correct - this is the enum that the codegen creates (a real thing of beauty! 🙄 )
```
    /**
     * How the match was performed
     *
     * Values: aLLSUPPLIED,aLLSUPPLIEDALIAS,hMPPSKEY,eXTERNALKEY,nAME,pARTIALNAME,pARTIALNAMEDOBLENIENT,nOTHING
     */
    enum class MatchedBy(val value: kotlin.String) {
        @JsonProperty(value = "ALL_SUPPLIED") aLLSUPPLIED("ALL_SUPPLIED"),
        @JsonProperty(value = "ALL_SUPPLIED_ALIAS") aLLSUPPLIEDALIAS("ALL_SUPPLIED_ALIAS"),
        @JsonProperty(value = "HMPPS_KEY") hMPPSKEY("HMPPS_KEY"),
        @JsonProperty(value = "EXTERNAL_KEY") eXTERNALKEY("EXTERNAL_KEY"),
        @JsonProperty(value = "NAME") nAME("NAME"),
        @JsonProperty(value = "PARTIAL_NAME") pARTIALNAME("PARTIAL_NAME"),
        @JsonProperty(value = "PARTIAL_NAME_DOB_LENIENT") pARTIALNAMEDOBLENIENT("PARTIAL_NAME_DOB_LENIENT"),
        @JsonProperty(value = "NOTHING") nOTHING("NOTHING");
    }
```